src/py/flwr/server/superlink/fleet/message_handler/message_handler.py

```diff
@@ -121,6 +132,15 @@ def get_run(
     if run is None:
         return GetRunResponse()
 
+    # Abort if the run is not running
+    abort_msg = check_abort(
+        request.run_id,
+        [Status.PENDING, Status.STARTING, Status.FINISHED],
```

Comment:
Shouldn't the SuperNode be able to get the run even if it's finished?

Reply:
Changed to `InvalidRunStatusException` to maintain consistent  `*Exception` naming.

Reply:
IMO, as far as the workflow goes now, the SuperNode should only be allowed (and able) to pull a Run that is running. At least this is what makes most sense to me.